src/pydvl/influence/torch/operator.py

```diff
@@ -329,3 +333,138 @@ def _apply_to_mat(self, mat: torch.Tensor) -> torch.Tensor:
+        regularization: Optional[float] = None,
+        maxiter: int = 1000,
+        dampen: float = 0.0,
+        scale: float = 10.0,
+        rtol: float = 1e-4,
+        progress: bool = False,
+        warn_on_max_iteration: bool = True,
+    ):
+
+        if regularization is not None and regularization < 0:
+            raise ValueError("regularization must be non-negative")
+
+        self.data = data
+        self.warn_on_max_iteration = warn_on_max_iteration
+        self.progress = progress
+        self.rtol = rtol
+        self.scale = scale
+        self.dampen = dampen
+        self.maxiter = maxiter
+        self.batch_operation = batch_operation
+        self._regularization = regularization
+
+    @property
+    def regularization(self):
+        return self._regularization
+
+    @regularization.setter
+    def regularization(self, value: float):
+        if value < 0:
+            raise ValueError("regularization must be non-negative")
+        self._regularization = value
+
+    @property
+    def device(self):
+        return self.batch_operation.device
+
+    @property
+    def dtype(self):
+        return self.batch_operation.dtype
+
+    def to(self, device: torch.device):
+        self.batch_operation = self.batch_operation.to(device)
+        return self
+
+    def _reg_apply(self, batch: TorchBatch, h: torch.Tensor):
+        result = self.batch_operation.apply(batch, h)
+        if self.regularization is not None:
+            result += self.regularization * h
+        return result
+
+    def _lissa_step(self, h: torch.Tensor, rhs: torch.Tensor, batch: TorchBatch):
+        result = rhs + (1 - self.dampen) * h - self._reg_apply(batch, h) / self.scale
+        if result.requires_grad:
+            result = result.detach()
+        return result
+
+    def _apply_to_vec(self, vec: torch.Tensor) -> torch.Tensor:
+        h_estimate = torch.clone(vec)
+        shuffled_training_data = DataLoader(
+            self.data.dataset,
+            self.data.batch_size,
+            shuffle=True,
+        )
+        for k in tqdm(
+            range(self.maxiter), disable=not self.progress, desc="Lissa iteration"
+        ):
+            x, y = next(iter(shuffled_training_data))
+
+            residual = self._lissa_step(h_estimate, vec, TorchBatch(x, y)) - h_estimate
+            h_estimate += residual
+            if torch.isnan(h_estimate).any():
+                raise RuntimeError("NaNs in h_estimate. Increase scale or dampening.")
+            max_residual = torch.max(torch.abs(residual / h_estimate))
+            if max_residual < self.rtol:
+                mean_residual = torch.mean(torch.abs(residual / h_estimate))
+                logger.debug(
+                    f"Terminated Lissa after {k} iterations with "
+                    f"{max_residual*100:.2f} % max residual and"
+                    f" mean residual {mean_residual*100:.5f} %"
+                )
+                break
+        else:
```

Comment:
The indentation of the `else` seems off. Shouldn't it be at the same level as the `if`s?

Reply:
Removed for/else construct, replaced with flag, done